### PR TITLE
TST: Test resampling with NaT

### LINKED ISSUE #13020: Unclear error produced by resampling with NaT in Index
Similar to #4746, trying to `resample` with an index containing `NaT` produces rather ambiguous error messages of the form: 

> ValueError: Shape of passed values is (2, 5), indices imply (2, 4)

Something like `cannot resample from index containing Not-a-Time (NaT)` would be more helpful to users. 

#### MWE 

```python
import pandas as pd
import numpy as np

times = pd.DatetimeIndex(['20160428T110000',
                          '20160428T110001',
                          '20160428T110002',
                          '20160428T110003',
                          '20160428T110004',
                          pd.NaT,
                          '20160428T110001',
                          '20160428T110002',
                          '20160428T110003',
                          '20160428T110004'])

df = pd.DataFrame(np.random.randn(10,2),
                  index=times,
                  columns=['A', 'B'])
df.resample('3s')
```

*This example index contains duplicate values, but results are the same w/ or w/o them.*

#### output of ``pd.show_versions()``

```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.9.final.0
python-bits: 32
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 23 Stepping 10, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.15.2
nose: 1.3.4
Cython: 0.21.1
numpy: 1.8.2
scipy: 0.14.0
statsmodels: 0.6.1
IPython: 2.3.1
sphinx: 1.2.3
patsy: 0.3.0
dateutil: 2.3
pytz: 2014.10
bottleneck: None
tables: 3.1.1
numexpr: 2.4
matplotlib: 1.4.2
openpyxl: None
xlrd: 0.9.3
xlwt: None
xlsxwriter: 0.6.4
lxml: 3.4.1
bs4: 4.3.2
html5lib: None
httplib2: None
apiclient: None
rpy2: 2.5.2
sqlalchemy: 0.9.8
pymysql: None
psycopg2: None
```

### PR DESCRIPTION
**Pull request by evectant**
- Closes #13020
- Adds `test_generic.py:TestDataFrame.test_resample_with_nat`
- Passes tests and flake8
- No documentation needed

### CODE REVIEW

**Comment by sinhrks** [comment]
Thanks. Can you also add `TimedeltaIndex` and `PeriodIndex`?

File: pandas/tests/test_generic.py [hunk 1]
```diff
@@ -1426,6 +1426,32 @@ def test_to_xarray(self):
                            expected,
                            check_index_type=False)
 
+    def test_resample_with_nat(self):
+        # GH 13020
+        index = pd.DatetimeIndex([pd.NaT,
```
Comment (jreback) [inline pandas/tests/test_generic.py]: move all of these to ``tests/tseries/test_resample.py``

**Comment by jreback** [comment]
@evectant can you rebase & update

**Comment by evectant** [comment]
@sinhrks Those don't seem to be very tolerant to NaT. I'm not sure what behavior we should pin down here.

```
In [2]: dti = pd.DatetimeIndex(['1970-01-01 00:00:00', pd.NaT, '1970-01-01 00:00:02'])

In [3]: pd.DataFrame([2, 3, 5], dti).resample('1s').mean()
Out[3]: 
                       0
1970-01-01 00:00:00  2.0
1970-01-01 00:00:01  NaN
1970-01-01 00:00:02  5.0
```

```
In [4]: tdi = pd.to_timedelta(['0s', pd.NaT, '2s'])

In [5]: pd.DataFrame([2, 3, 5], tdi).resample('1s').mean()
...
MemoryError:
```

```
In [6]: pi = pd.PeriodIndex(['1970-01-01 00:00:00', pd.NaT, '1970-01-01 00:00:02'], freq='S')

In [7]: pd.DataFrame([2, 3, 5], pi).resample('1s').mean()
...
ValueError: Start and end must not be NaT
```

File: pandas/tseries/tests/test_resample.py [hunk 2]
```diff
@@ -2839,6 +2839,32 @@ def test_aggregate_with_nat(self):
             # if NaT is included, 'var', 'std', 'mean', 'first','last'
             # and 'nth' doesn't work yet
 
+    def test_resample_with_nat(self):
```
Comment (jreback) [inline pandas/tseries/tests/test_resample.py]: if you move these to ``Base`` and use ``.create_series()`` you will get testing for free for all of the sub-classes (DTI, PeriodIndex, TimedeltaIndex)

**Comment by evectant** [comment]
@jreback Would it be helpful here? All three index types seem to handle NaT differently (see examples above).

File: pandas/tseries/tests/test_resample.py (hunk 2 above)
Comment (evectant) [inline pandas/tseries/tests/test_resample.py]: Would it be helpful here? All three index types seem to handle NaT differently (see examples above).

**Comment by sinhrks** [comment]
@evectant Thx to confirm. It's gr8 if you can fix, otherwise keep #13020 open or submit new issue for tracking.

**Comment by evectant** [comment]
@sinhrks @jreback I think there are three different issues here:
- Resampling with `NaT` in `DatetimeIndex` raises `ValueError` (fixed in master, pinned down here)
- Resampling with `NaT` in `TimedeltaIndex` raises `MemoryError` (entered #13223, will look into)
- Resampling with `NaT` in `PeriodIndex` raises `ValueError` (entered #13224, will look into)

So can this one be merged and #13020 closed?

**Comment by jreback** [comment]
thanks @evectant 

merged.

when you work on the other too resample-with-nat issues, ideally we move this test to ``Base`` (so it cycles thru all index types); then these are all in 1 place.

**Pull request closed without merging**